### DISCUSSION TITLE
v3.1.0

### DISCUSSION THREAD

**Discussion by leonrayang** [cubefs/cubefs#1524] [Announcements]
### **UPGRAGDE NOTICE**
If your CubeFS version is v2.3.x or before, please refer to the UPGRADE NOTICE in v2.4.0 for upgrading steps. And also please make sure that your fuse client or objectnode version is equal to or older than the servers, i.e. master, metanode and datanode. In another word, newer versioned client can not be used in a cluster with older versioned servers.

### **Main Feature**
* `master\client\datanode`: ：Provide QoS services to enhance multi-tenant isolation #1447
* `client`: Caching acceleration #1446
* `master\datanode`: Support two replicas data storage #1389

### **Bugfix**
* `master`: Add vol label in cfs_metanode_mpInodeCount #1482
* `client`: Support config posixAcl for volmue #1485
* `master`: Add datapartition count limit for datanode. #1481
* `datanode`: TinyDeleteRecord file will be very big when one replica is down #1433
* `client`: EnablePosixACL is not vaild in centos system #1393
* `bcache`: Bug report cfs-bcache memory overflow and be killed #1422
* `yum source`: Yum install script fails due to missing packages #1410
* `unit test`: S3test random path busy to produce data and timeout #1515
* `master`: Zone name check not strict enough while crossZone enabled #1479
* `datanode\master`: Avoid writing operation failed because of disk full #1519

### **Enhance**
* `master`: enable volume update from 3 to 2 replics  #1516

 

<hr /><em>This discussion was created from the release <a href='https://github.com/cubefs/cubefs/releases/tag/v3.1.0'>v3.1.0</a>.</em>

**Comment by pengtianyue025** [comment]
NB
